File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        unordered_set<char> st;
        int left = 0, right = 0;

        int ans = 0;

        while(right < s.size()){

            while(!st.empty() && st.find(s[right]) != st.end()){
                st.erase(s[left++]);
            }

            st.insert(s[right]);
            ans = max(ans, right - left + 1);
            right++;

        }

        return ans;
        
    }
};

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        Set<Character> set = new HashSet<>();
        int maxLen = 0;
        int left = 0, right = 0;
        while(right < s.length()){

            while(set.contains(s.charAt(right))){
                set.remove(s.charAt(left++));
            }

            maxLen = Math.max(maxLen, right - left + 1);
            set.add(s.charAt(right));
            right++;
        }
        return maxLen;
    }
}